packages/mui-base/src/SwitchUnstyled/SwitchUnstyled.tsx

```diff
@@ -124,11 +133,39 @@ const SwitchUnstyled = React.forwardRef(function SwitchUnstyled<
     className: classes.track,
   });
 
+  const handleOnClick = (event: React.BaseSyntheticEvent) => {
+    setCheckedState(!checked);
+    setFocusVisible(false);
+    const nativeEvent = event.nativeEvent || event;
+    // @ts-ignore
+    const clonedEvent = new nativeEvent.constructor(nativeEvent.type, nativeEvent);
+    Object.defineProperty(clonedEvent, 'target', inputRef.current);
+    onChange?.(clonedEvent);
+  };
+
+  const handleOnKeyPress = (e: React.KeyboardEvent) => {
+    if (e.code === 'Space' && inputRef.current) {
+      setCheckedState(!checked);
+    }
+  };
+
+  const handleOnFocus = () => {
+    setFocusVisible(true);
+  };
```

Comment:
Ideally, these event handlers should be specified in `useSwitch`, so they won't have to be reimplemented in userland each time someone uses out hook. Plus, this will allow not to export the `setCheckedState`, as the API should be declarative if only possible (`value` / `defaultValue` instead of `setValue`).